### ISSUE TITLE
Linux headless server does not export properly in docker container

### ISSUE THREAD

**Issue by lelandhwu** [godotengine/godot#59285]
### Godot version

3.3.4+

### System information

Mac OS Monterey for host machine, Alpine Linux for Docker

### Issue description

Hi! I'm trying to export a Mac OSX export from the linux headless build on docker based off of this repo:  https://github.com/GodotNuts/GodotServer-Docker/blob/main/auto-export/dockerfile

the image that i have on my public registry is basically a copy and paste of this project minus some:
https://gist.github.com/lelandhwu/bd21f0a0faecda70de87f3ce061aa2a5

it works great, the headless server works great for multiplayer. the only thing is for my CI/CD builds when i try to export a mac osx version using the following command: godot --export "Mac OSX" file.dmg, it becomes corrupted. I've tested this by spinning up a local docker container running headless godot, exporting the file, and then running a docker cp to get the file back onto my host machine. I've also tested this with a fresh project with nothing on it. The .dmg file still gets corrupted

any ideas? next thing im thinking about is bumping the godot version on my public registry. I'd love some ideas from you guys though. 

Thanks so much! 

### Steps to reproduce

1. Create a doctor image with the following code: https://gist.github.com/lelandhwu/bd21f0a0faecda70de87f3ce061aa2a5
we've tried version 3.3.4. The highest we'll go is 3.4.3 for now 

2. Push it up to a registry. We'll be using this as a base image for our future builds.
 
3. Create a simple godot project with a main scene. We'll use this to test. Create a Dockerfile in the root directory of tha tproject that looks like this: https://gist.github.com/lelandhwu/b8233f352c616b23ff0a6325189a14f7
This is the other part of the file, just in a separate docker image

4. build the docker image:  docker build . -f Dockerfile
5. run the image in a separate process: docker run -it image-id /bin/bash
6. In a separate process, cp the exported the artifact onto your host machine. docker cp container-id:/godotapp/hz.dmg .
7. try and run the dmg file

### Minimal reproduction project

[test_project.zip](https://github.com/godotengine/godot/files/8306058/test_project.zip)

**Comment by bruvzg** [comment]
> godot --export "Mac OSX" file.dmg

Exporting as `.dmg` is supported only on macOS host, on Linux hosts is's possible to export macOS app only as a `.zip`.

But even if it's exported as `.zip`, the application will not be signed, it will be blocked by Gatekeeper on all Macs and won't run on M1 powered Macs at all. 3.5 and 4.0 will have ad-hoc singing support on Linux, but proper singing with the Apple issued signature and notarization is still will be only available on macOS hosts.

**Comment by lelandhwu** [comment]
is there any relatively simple way to export a signed Mac OS version from a headless Godot server for ci/cd pipelines?

**Comment by bruvzg** [comment]
Only if your CI/CD is running on macOS (there's no official binaries, but it should be possible to build a headless version for macOS, or use normal macOS editor binary instead).

**Comment by Calinou** [comment]
We should modify the exporting code to print an error when attempting to export a `.dmg` on a non-macOS platform, including in headless mode.

**Comment by lelandhwu** [comment]
Any pointers for running headless Mac binaries?

**Comment by Calinou** [comment]
> Any pointers for running headless Mac binaries?

In `3.x`, you need to [compile a binary](https://docs.godotengine.org/en/stable/development/compiling/compiling_for_osx.html) with `platform=server` on a macOS machine, and it'll produce a headless macOS editor binary.

**Comment by jamie-pate** [comment]
According to wwdc2022 you can now notarize by REST api! https://developer.apple.com/videos/play/wwdc2022/10109/?time=90

**Comment by fire** [comment]
Duplicate of https://github.com/godotengine/godot-proposals/issues/5115 but for Godot Engine 3

**Comment by bruvzg** [comment]
> According to wwdc2022 you can now notarize by REST api!

In 4.0 you can sign and notarize macOS apps on Linux/Windows using [PyOxidizer rccodesign](https://github.com/indygreg/PyOxidizer/tree/main/apple-codesign), see https://github.com/godotengine/godot/pull/64207

**Comment by bruvzg** [comment]
> but for Godot Engine 3

There's nothing 4.x specific, so should be possible to backport it for 3.6.
